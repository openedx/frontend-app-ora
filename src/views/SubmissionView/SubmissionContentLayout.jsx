import React from 'react';
import PropTypes from 'prop-types';

import { Col, Row } from '@edx/paragon';

import Rubric from 'components/Rubric';
import SubmissionContent from './SubmissionContent';

import './SubmissionContentLayout.scss';

const SubmissionContentLayout = ({
  submission,
  oraConfigData,
  onTextResponseChange,
  onFileUploaded,
  draftSaved,
}) => (
  <div className="assessment-content-layout mr-auto ml-auto">
    <div className="content-wrapper">
      <Row className="flex-nowrap m-0">
        <Col className="p-0">
          <SubmissionContent
            submission={submission}
            oraConfigData={oraConfigData}
            onTextResponseChange={onTextResponseChange}
            onFileUploaded={onFileUploaded}
            draftSaved={draftSaved}
          />
        </Col>
        {oraConfigData.showDuringResponse && <Rubric isGrading={false} />}
      </Row>
    </div>
  </div>
);

SubmissionContentLayout.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  submission: PropTypes.any.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  oraConfigData: PropTypes.any.isRequired,
  onTextResponseChange: PropTypes.func.isRequired,
  onFileUploaded: PropTypes.func.isRequired,
  draftSaved: PropTypes.bool.isRequired,
};

export default SubmissionContentLayout;
