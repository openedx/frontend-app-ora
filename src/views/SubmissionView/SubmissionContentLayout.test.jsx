import { shallow } from '@edx/react-unit-test-utils';
import SubmissionContentLayout from './SubmissionContentLayout';

jest.mock('components/Rubric', () => 'Rubric');
jest.mock('./SubmissionContent', () => 'SubmissionContent');

describe('<SubmissionContentLayout />', () => {
  const props = {
    submission: 'submission',
    oraConfigData: {
      showDuringResponse: true,
    },
    onTextResponseChange: jest.fn().mockName('onTextResponseChange'),
    onFileUploaded: jest.fn().mockName('onFileUploaded'),
    draftSaved: true,
  };

  it('show rubric', () => {
    const wrapper = shallow(<SubmissionContentLayout {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('Rubric')).toHaveLength(1);
  });

  it('hide rubric', () => {
    const wrapper = shallow(<SubmissionContentLayout {...props} oraConfigData={{ showDuringResponse: false }} />);
    expect(wrapper.snapshot).toMatchSnapshot();

    expect(wrapper.instance.findByType('Rubric')).toHaveLength(0);
  });
});
