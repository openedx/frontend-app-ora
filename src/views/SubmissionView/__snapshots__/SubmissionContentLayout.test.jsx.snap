// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmissionContentLayout /> hide rubric 1`] = `
<div
  className="assessment-content-layout mr-auto ml-auto"
>
  <div
    className="content-wrapper"
  >
    <Row
      className="flex-nowrap m-0"
    >
      <Col
        className="p-0"
      >
        <SubmissionContent
          draftSaved={true}
          onFileUploaded={[MockFunction onFileUploaded]}
          onTextResponseChange={[MockFunction onTextResponseChange]}
          oraConfigData={
            Object {
              "showDuringResponse": false,
            }
          }
          submission="submission"
        />
      </Col>
    </Row>
  </div>
</div>
`;

exports[`<SubmissionContentLayout /> show rubric 1`] = `
<div
  className="assessment-content-layout mr-auto ml-auto"
>
  <div
    className="content-wrapper"
  >
    <Row
      className="flex-nowrap m-0"
    >
      <Col
        className="p-0"
      >
        <SubmissionContent
          draftSaved={true}
          onFileUploaded={[MockFunction onFileUploaded]}
          onTextResponseChange={[MockFunction onTextResponseChange]}
          oraConfigData={
            Object {
              "showDuringResponse": true,
            }
          }
          submission="submission"
        />
      </Col>
      <Rubric
        isGrading={false}
      />
    </Row>
  </div>
</div>
`;
